### PR TITLE
SONARGITUB-23 Update to SQ plugin API 5.6 (Java 8)

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java [hunk 1]
```diff
@@ -73,52 +67,38 @@ public void executeOn(Project project, SensorContext context) {
...
+      .filter(i -> i.isNew())
+      // SONARGITUB-13 Ignore issues on files not modified by the P/R
+      .filter(i -> {
+        InputComponent inputComponent = i.inputComponent();
+        return inputComponent == null ||
+          !inputComponent.isFile() ||
+          pullRequestFacade.hasFile((InputFile) inputComponent);
+      })
+      .sorted(ISSUE_COMPARATOR)
+      .forEach(i -> {
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java]: MAJOR Remove useless curly braces around statement [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1602)
MINOR Replace this lambda with a method reference. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1612)

File: src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java [hunk 2]
```diff
@@ -73,52 +67,38 @@ public void executeOn(Project project, SensorContext context) {
...
   }
 
-  private void processIssue(GlobalReport report, Map<InputFile, Map<Integer, StringBuilder>> commentToBeAddedByFileAndByLine, Issue issue, String severity,
-    @Nullable Integer issueLine,
-    @Nullable InputFile inputFile) {
+  private void processIssue(GlobalReport report, Map<InputFile, Map<Integer, StringBuilder>> commentToBeAddedByFileAndByLine, PostJobIssue issue) {
     boolean reportedInline = false;
-    if (gitHubPluginConfiguration.tryReportIssuesInline()) {
-      reportedInline = tryReportInline(commentToBeAddedByFileAndByLine, issue, severity, issueLine, inputFile);
+    if (gitHubPluginConfiguration.tryReportIssuesInline() && issue.inputComponent() != null && issue.inputComponent().isFile()) {
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java]: BLOCKER NullPointerException might be thrown as 'inputComponent' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

File: src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java [hunk 3]
```diff
@@ -73,52 +67,38 @@ public void executeOn(Project project, SensorContext context) {
...
-        // SONARGITUB-13 Ignore issues on files not modified by the P/R
-        continue;
-      }
-      processIssue(report, commentToBeAddedByFileAndByLine, issue, severity, issueLine, inputFile);
-    }
+    StreamSupport.stream(issues.spliterator(), false)
+      .filter(i -> i.isNew())
+      // SONARGITUB-13 Ignore issues on files not modified by the P/R
+      .filter(i -> {
+        InputComponent inputComponent = i.inputComponent();
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java]: MAJOR Make this line start at column 13. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* BLOCKER 3 blocker
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. BLOCKER [MarkDownUtils.java#L44](https://github.com/SonarSource/sonar-github/blob/72383cf2f3b68cd030babe1b21ec9cebf13ad126/src/main/java/org/sonar/plugins/github/MarkDownUtils.java#L44): NullPointerException might be thrown as 'baseUrl' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)
1. BLOCKER [MarkDownUtils.java#L44](https://github.com/SonarSource/sonar-github/blob/72383cf2f3b68cd030babe1b21ec9cebf13ad126/src/main/java/org/sonar/plugins/github/MarkDownUtils.java#L44): NullPointerException might be thrown as 'baseUrl' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 2 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. BLOCKER [MarkDownUtils.java#L44](https://github.com/SonarSource/sonar-github/blob/77ccf5d75eeaacdbd61fc5f9683c38f86d52fa80/src/main/java/org/sonar/plugins/github/MarkDownUtils.java#L44): NullPointerException might be thrown as 'baseUrl' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)
1. BLOCKER [MarkDownUtils.java#L44](https://github.com/SonarSource/sonar-github/blob/77ccf5d75eeaacdbd61fc5f9683c38f86d52fa80/src/main/java/org/sonar/plugins/github/MarkDownUtils.java#L44): NullPointerException might be thrown as 'baseUrl' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 2 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. BLOCKER [MarkDownUtils.java#L44](https://github.com/SonarSource/sonar-github/blob/07b7d65c60596f04f5725e200a6c81fdfc05d260/src/main/java/org/sonar/plugins/github/MarkDownUtils.java#L44): NullPointerException might be thrown as 'baseUrl' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)
1. BLOCKER [MarkDownUtils.java#L44](https://github.com/SonarSource/sonar-github/blob/07b7d65c60596f04f5725e200a6c81fdfc05d260/src/main/java/org/sonar/plugins/github/MarkDownUtils.java#L44): NullPointerException might be thrown as 'baseUrl' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

File: src/main/java/org/sonar/plugins/github/PullRequestFacade.java [hunk 4]
```diff
@@ -270,6 +271,13 @@ public void createOrUpdateSonarQubeStatus(GHCommitState status, String statusDes
         targetUrl = lastStatus.getTargetUrl();
       }
       ghRepo.createCommitStatus(pr.getHead().getSha(), status, targetUrl, statusDescription, COMMIT_CONTEXT);
+    } catch (FileNotFoundException e) {
+      String msg = "Unable to set pull request status. GitHub account probably miss push permission on the repository.";
+      if (LOG.isDebugEnabled()) {
+        LOG.warn(msg, e);
```
Comment (simonbrandhof) [inline src/main/java/org/sonar/plugins/github/PullRequestFacade.java]: I suppose it should be LOG.debug()

**Comment by simonbrandhof** [comment]
LGTM

**Pull request merged**